rules-reviewed/eap7/eap6/hsearch.windup.xml

```diff
@@ -0,0 +1,1197 @@
+                        title="Advanced Integrator Changes in Hibernate Search 5.x" />
+                    <link href="http://docs.jboss.org/hibernate/search/5.5/api/org/hibernate/search/spi/SearchIntegratorBuilder.html" title="SearchIntegrationBuilder javadoc" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00010">
+            <when>
+                <javaclass references="org.hibernate.search.spi.SearchFactoryIntegrator" />
+            </when>
+            <perform>
+                <hint title="Deprecated interface org.hibernate.search.spi.SearchFactoryIntegrator in Hibernate Search 5" effort="1" severity="mandatory">
+                    <message>Do not use `org.hibernate.search.spi.SearchFactoryIntegrator`. You should migrate all code to use `org.hibernate.search.spi.SearchIntegrator`.
+                    This interface is a temporary placeholder and will be removed at the next micro release!</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_hibernate_search_advanced_integrator_changes"
+                        title="Advanced Integrator Changes in Hibernate Search 5.x" />
+                    <link href="https://docs.jboss.org/hibernate/search/5.5/api/org/hibernate/search/spi/SearchIntegrator.html"
+                        title="Hibernate Search javadoc for SearchIntegrator" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00011">
+            <when>
+                <javaclass references="org.hibernate.search.Version" />
+            </when>
+            <perform>
+                <hint title="Hibernate Search 5 package change for Version" effort="1" severity="mandatory">
+                    <message>The class `org.hibernate.search.Version` was replaced by `org.hibernate.search.engine.Version` class.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_hibernate_search_changes"
+                        title="Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <!-- Deprecated things -->
+        <rule id="hsearch-00100">
+            <when>
+                <javaclass references="org.hibernate.search.cfg.NumericFieldMapping(org.hibernate.search.cfg.PropertyDescriptor, org.hibernate.search.cfg.EntityDescriptor, org.hibernate.search.cfg.SearchMapping)">
```

Comment:
I wonder how much is this related to rule `hsearch-00211`. I mean, I don't see a problem in keeping this rule, but it seems not very useful.